### DISCUSSION TITLE
Pilot Share Code expiration -- design intent question

### DISCUSSION THREAD

**Discussion by sealtrowitz** [massif-press/compcon#2082] [Archive]
I realized a few days ago that all my player's share codes had expired, and that's why they weren't working anymore. Out of curiosity, what's the intention/purpose/reason behind pilot share codes having an expiration? My best guess it has something to do with saving bandwidth or costs incurred by the host?

**Comment by msprijatelj** [comment]
I believe it has to do with costs of maintaining records, yes. Ideally, you should be able to refresh the existing share code before it expires without needing to regenerate it? But I believe there may be a bug preventing a share code refresh at this time.

**Reply by sealtrowitz** [reply]
That makes sense. One of my players who works in programming/development was confused by it because it appears to save the code even after the expiration, which, according to him, would mean it "still has to ping to check the share code expiration, even when it's expired" and thus would still use bandwidth, just slightly less than sending the full pilot data. I admittedly don't know enough to comment on it beyond mentioning to him that I thought it might be for cost purposes, so thanks for confirming!

Re: the share code refresh, I've noticed that refreshing the share code is only possible to do once the expiration date has passed, so trying to refresh before then just has the option greyed out, though I had thought that was intentional.

**Reply by msprijatelj** [reply]
I tested the share code refresh today: The refresh button fired, it says it's already at maximum, but when I attempt to import the pilot, I receive a "Record found! Attempting to load data..." followed by a 404 not found error for the S3 retrieval. Regenerating the code completely allows for a successful import, as a workaround.

**Reply by sealtrowitz** [reply]
Ah yeah, just confirmed that refreshing a pilot's share code _appears_ to work but doesn't allow for imports into Foundry and as a remote resource, etc. Regenerating the code is what I'll do for now, and have my players do. Thanks!

Should I make an issue regarding pilot share code refreshing?

EDIT: Nevermind, just saw you already did, haha. Nice!

**Reply by jarena3** [reply]
Thanks for bringing this up, refreshing the code should be sufficient so there's definitely something going on there I need to clean up. Also refreshing the code not working until the exp date has passed is bad practice, I should reduce that to something like 24 or 72h maybe? Just something so it can't be spammed.

To your player, though: I save that (un)expired timer with the client data, so I can check against timestamps locally rather than having to check with the backend every time to see if the code is stale or not -- but yeah it is a cost-saving measure (a lot of the cloud-based weirdness is, it'd be much easier, reliable, and more resilient to just manage everything in the cloud but that gets expensive really fast!)

**Reply by sealtrowitz** [reply]
Yeah, I think putting a timer on it would be a smart move. And I let him know what you said -- he said that makes sense!
